src/m365/planner/commands/bucket/bucket-remove.ts

```diff
@@ -0,0 +1,255 @@
+    return ['bucketId', 'bucketName', 'planId', 'confirm'];
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: (err?: any) => void): void {
+    const removeBucket: () => void = (): void => {
+      if (args.options.bucketName) {
+
+        this.getPlanId(args)
+          .then((planId: string): Promise<any> => {
+            const requestOptions: any = {
+              url: `${this.resource}/v1.0/planner/plans/${planId}/buckets`,
+              headers: {
+                'accept': 'application/json',
+                'prefer': 'odata.eTag'
+              },
+              responseType: 'json'
+            };
+            return request.get<{ value: { id: string; title: string; }[] }>(requestOptions);
+
+          }).then((response) => {
+            const bucket: Bucket | undefined = response.value.find((bucket: Bucket) => bucket.name === args.options.bucketName);
+            if (!bucket) {
+              return Promise.reject(`The specified bucket does not exist in the Microsoft Planner`);
+            }
+            const requestOptions: any = {
+              url: `${this.resource}/v1.0/planner/buckets/${bucket.id}`,
+              headers: {
+                'X-HTTP-Method': 'DELETE',
+                'If-Match': bucket['@odata.etag'],
+                'accept': 'application/json;odata=nometadata'
+              },
+              responseType: 'json'
+            };
+            return request.delete(requestOptions);
+          }).then(_ => cb(), (err: any) => {
+            this.handleRejectedODataJsonPromise(err, logger, cb);
+          });
+      }
+      if (args.options.bucketId) {
+        this.getBucket(args)
+          .then((res: Bucket): Promise<void> => {
+            const bucketid: string = res.id;
+            const requestOptions: any = {
+              url: `${this.resource}/v1.0/planner/buckets/${bucketid}`,
+              headers: {
+                'X-HTTP-Method': 'DELETE',
+                'If-Match': res['@odata.etag'],
+                'accept': 'application/json;odata=nometadata'
+              },
+              responseType: 'json'
+            };
+            return request.delete(requestOptions);
+          })
+          .then(_ => cb(), (err: any) => {
+            this.handleRejectedODataJsonPromise(err, logger, cb);
+          });
+      }
+    };
+    if (args.options.confirm) {
+      removeBucket();
+    }
+    else {
+      const bucketName = args.options.bucketName ? args.options.bucketName : args.options.bucketId;
+      Cli.prompt({
+        type: 'confirm',
+        name: 'continue',
+        default: false,
+        message: `Are you sure you want to remove the bucket ${bucketName} from planner?`
+      }, (result: { continue: boolean }): void => {
+        if (!result.continue) {
+          cb();
+        }
+        else {
+          removeBucket();
+        }
+      });
+    }
+  }
+
+  private getBucket(args: CommandArgs): Promise<any> {
```

Comment:
Since we know the type of data returned by MS Graph, let's not use `any` as the return type, but rather use the Bucket type to benefit from design-time typesafety features.